multiqc/modules/seqkit/stats.py

```diff
@@ -0,0 +1,394 @@
+                "title": "SeqKit: Stats",
+                "namespace": "seqkit",
+            },
+        ),
+    )
+
+    # Create bar plot for sequence counts and lengths
+    bargraph_data_seqs = {s_name: {"Sequences": d.get("num_seqs", 0)} for s_name, d in seqkit_stats.items()}
+
+    bargraph_data_bases = {s_name: {"Total Bases": d.get("sum_len", 0)} for s_name, d in seqkit_stats.items()}
+
+    module.add_section(
+        name="Sequence Counts",
+        anchor="seqkit-stats-seqcounts",
+        description="Number of sequences per sample.",
+        plot=bargraph.plot(
+            bargraph_data_seqs,
+            pconfig={
+                "id": "seqkit-stats-seq-counts",
+                "title": "SeqKit: Sequence Counts",
+                "ylab": "Number of Sequences",
+            },
+        ),
+    )
+
+    module.add_section(
+        name="Total Bases",
+        anchor="seqkit-stats-bases",
+        description="Total number of bases per sample.",
+        plot=bargraph.plot(
+            bargraph_data_bases,
+            pconfig={
+                "id": "seqkit-stats-total-bases",
+                "title": "SeqKit: Total Bases",
+                "ylab": "Number of Bases",
+            },
+        ),
+    )
+
+    # Write parsed report data to a file
+    module.write_data_file(seqkit_stats, "multiqc_seqkit_stats")
+
+    return len(seqkit_stats)
+
+
+def parse_stats_report(file_content: str) -> Dict[str, Dict]:
+    """
+    Parse seqkit stats output file
+
+    Returns a dictionary with sample names as keys and parsed data as values.
+    Handles both tab-separated (--tabular flag) and space-separated (default) output.
+    """
+    parsed_data: Dict[str, Dict] = {}
+    lines = file_content.strip().split("\n")
+
+    if len(lines) < 2:
+        return {}
+
+    # Parse header line - try tab-separated first, then space-separated
+    header_line = lines[0]
+    if "\t" in header_line:
+        headers = [h.strip() for h in header_line.split("\t")]
+        use_tabs = True
+    else:
+        headers = header_line.split()
+        use_tabs = False
+
+    # Check if this looks like seqkit stats output
+    expected_headers = ["file", "format", "type", "num_seqs", "sum_len", "min_len", "avg_len", "max_len"]
+    if not all(h in headers for h in expected_headers):
+        return {}
+
+    # Parse data lines
+    for line in lines[1:]:
+        if not line.strip():
+            continue
+
+        if use_tabs:
+            values = line.split("\t")
+        else:
+            values = line.split()
+        if len(values) < len(headers):
+            continue
+
+        data: Dict = {}
+        sample_name = None
+
+        for i, header in enumerate(headers):
+            value = values[i].strip() if i < len(values) else ""
+
+            if header == "file":
+                # Use file name as sample name, strip path and extension
+                if value == "-":
+                    sample_name = "stdin"
```

Comment:
This will mean that only one file can ever be parsed in this mode.

Instead, use `f['fn']`